### DISCUSSION TITLE
Disable auto fix "Remove unused function" on format/save

### DISCUSSION THREAD

**Discussion by Akronae** [OmniSharp/omnisharp-vscode#6260] [Q&A]
I sometimes have this warning
```
The local function 'xxx' is declared but never used csharp(CS8321)
```

I'd like the editor not to delete `xxx` on format/save, although it is unused.
I tried adding
```
roslynator_compiler_diagnostic_fix.CS8321.enabled = false
```
To my `.editorconfig` but the editor is still automatically deleting my code!
Any idea how to tackle that?

My `settings.json`
```json
  "[csharp]": {
    "editor.tabSize": 4,
    "editor.defaultFormatter": "csharpier.csharpier-vscode",
    "editor.codeActionsOnSave": [
      "source.organizeImports",
      "source.fixAll",
      "source.sortMembers",
    ],
  },
  "editor.formatOnSave": true,
  "omnisharp.enableRoslynAnalyzers": true,
  "omnisharp.organizeImportsOnFormat": true,
  "omnisharp.dotnetPath": "/usr/local/share/dotnet/dotnet",
  "omnisharp.sdkPath": "/usr/local/share/dotnet/sdk/7.0.306",
  "csharp.referencesCodeLens.enabled": false,
```

**Comment by JoeRobich** [comment]
Can you please tell us more about your configuration? Are you using the new Roslyn LSP or have you configured the extension to use OmniSharp? What actions have your configured VS Code to run on format/save?


I would not expect CodeFixes to be applied on save unless you are using the OmniSharp language service with fixAll on save configured.
```
"editor.codeActionsOnSave": {
  "source.fixAll.csharp": true
}
```

**Reply by Akronae** [reply]
Hi @JoeRobich, yes I'm asking VSCode to format/fix on save, and I do have `source.fixAll`, I should have posted my settings sorry, just edited.
I'm looking for a way to remove `CS8321` from the fixes as I consider it dangerous to automaticaly remove code!
